app/components/market/submit/parameter-input.component.ts

```diff
@@ -52,14 +52,27 @@ export class ParameterInputComponent implements ControlValueAccessor, OnChanges,
     }
 
     public validate() {
-       const valid = this.parameterValue.valid;
-       if (valid) { return null; }
-       return {
-            validJsonConfig: {
-                valid: false,
-                message: "FormControl is invalid",
-            },
-        };
+        const valid = this.parameterValue.valid;
+        if (valid) {
+            return null;
+        } else {
+            let messageText = "unknown error";
+            if (this.parameterValue.errors.minlength) {
+                messageText = "FormControl minLength error";
```

Comment:
use the message provided don't rewrite it. Here you are losing information.

Reply:
will add more detail to error message

Reply:
done